src/compiler/checker.ts

```diff
@@ -7163,17 +7163,48 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 }
             }
 
-            const modifiers = isReadonlySymbol(propertySymbol) ? [factory.createToken(SyntaxKind.ReadonlyKeyword)] : undefined;
-            if (modifiers) {
-                context.approximateLength += 9;
+            if (propertySymbol.flags & SymbolFlags.Accessor) {
+                const modifierFlags = getDeclarationModifierFlagsFromSymbol(propertySymbol);
+                const flags = modifierFlags & ~(ModifierFlags.Async | ModifierFlags.Static | ModifierFlags.Accessor);
+                if (propertySymbol.flags & SymbolFlags.GetAccessor) {
+                    const getAccessorSignature = factory.createGetAccessorDeclaration(
+                        factory.createModifiersFromModifierFlags(flags),
+                        propertyName,
+                        [],
+                        propertyTypeNode,
+                        /*body*/ undefined);
+                    typeElements.push(preserveCommentsOn(getAccessorSignature));
+                }
+                if (propertySymbol.flags & SymbolFlags.SetAccessor) {
+                    const setAccessorDecl = find(propertySymbol.declarations, decl => decl.kind === SyntaxKind.SetAccessor) as SetAccessorDeclaration;
+                    const parameterName = setAccessorDecl?.parameters?.length > 0 ? setAccessorDecl.parameters[0].name : "arg";
+                    const setAccessorSignature = factory.createSetAccessorDeclaration(
+                        factory.createModifiersFromModifierFlags(flags),
+                        propertyName,
+                        [factory.createParameterDeclaration(
+                            /*modifiers*/ undefined,
+                            /*dotDotDotToken*/ undefined,
+                            parameterName,
+                            /*questionToken*/ undefined,
+                            propertyTypeNode
+                        )],
+                        /*body*/ undefined);
+                    typeElements.push(preserveCommentsOn(setAccessorSignature));

```

Comment:
So, as-is, by using `preserveCommentsOn` in both the `get` and `set` branches, this'll duplicate comments from the `valueDeclaration` of the symbol onto both the emitted `get` and `set` accessors. You probably wanna do something a bit more bespoke that lookups up the get/set accessor declarations and copies the comments from that specific matching declaration kind. `setCommentRange(setAccessorSignature, setAccessorDecl)` should do.

Reply:
Thanks @weswigham. I've made the change you suggested, and also added a test for it.